File: result/metagpt/claude/humaneval/72.c
#include <stdbool.h>

bool will_it_fly(int q[], int size, int w) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        if (q[i] != q[size - 1 - i]) return false;
        sum += q[i];
    }
    if (sum > w) return false;
    return true;
}

File: result/metagpt/deepseek/humaneval/72.rs
pub fn will_it_fly(q: &[i32], w: i32) -> bool {
    // 检查是否对称
    if !q.iter().eq(q.iter().rev()) {
        return false;
    }
    
    // 检查总和是否超过阈值
    q.iter().sum::<i32>() <= w
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_will_it_fly() {
        assert!(will_it_fly(&[1, 2, 1], 5));
        assert!(!will_it_fly(&[1, 2, 3], 5));
        assert!(!will_it_fly(&[1, 2, 2, 1], 4));
    }
}